### DISCUSSION TITLE
Initialize a DataEntry without date

### DISCUSSION THREAD

**Discussion by fmerchan65** [israel-dryer/ttkbootstrap#476] [Q&A] [answered]
I don't see the way to initialize a DateEntry without date. If you clean it, today date appears. Why don't have a method to clean o set a specific date ?

I tried using "date.entry. insert (0, tb.END)" but it is not working. Also with date.configure(startdate=None) but nothing happen.

**Comment by israel-dryer** [comment]
Thanks for raising this. I reviewed the current v1 implementation, and this is still a limitation.

`DateEntry` currently treats an omitted or `None` `startdate` as today and immediately writes that date into the entry. Clearing the underlying entry text manually is not a complete workaround because the widget can still retain and return its internal date value.

I’ve created two tracked implementation issues:

* v1, backward-compatible implementation: #1252
* v2, revised nullable-value model: #1253

For v1, the recommendation is to preserve the existing today-by-default behavior while adding an opt-in `initially_empty=True` parameter, plus `clear()` and support for `set_date(None)`.

For v2, the recommendation is to make `value` the nullable selected date and keep `start_date` separate as the date used to position the popup when no value is selected. This avoids using the popup date as a hidden fallback value.

Until the v1 change is implemented, manually deleting the text from `date_entry.entry` only clears the display and should not be relied upon as a supported way to clear the widget’s value.

**Comment by israel-dryer** [comment, marked as answer]
This is resolved as of **ttkbootstrap 2.1.0** (and available in the current 2.2.0 release).

The shipped API is not the `initially_empty=True` parameter I proposed above — it's the nullable-value model from #1253. `DateEntry` now separates the **selected value** from the **popup start date**, so an empty field is a real state rather than a display trick.

Pass `value=None` to construct an empty field:

```python
import ttkbootstrap as ttk

entry = ttk.DateEntry(app, value=None)
entry.entry.get()   # ''
entry.get_date()    # None
```

`clear()` and `set_date(None)` are the supported ways to empty it, and the internal date no longer comes back:

```python
entry.set_date(date(2026, 1, 15))
entry.get_date()    # datetime(2026, 1, 15, 0, 0)

entry.clear()       # or entry.set_date(None)
entry.get_date()    # None
```

`value` is also a property, so `entry.value = None` and `entry.value` work the same way.

Omitting `value` entirely keeps the existing behavior — `DateEntry()` still selects today — so nothing changes for current code. Passing `value` (including `None`) is what opts into the nullable model. `start_date` now only positions the popup when nothing is selected; it is never used as a hidden fallback value.

Making `value=None` the **default**, so a bare `DateEntry()` starts empty, is a breaking change and is tracked separately for 3.0 in #1276.
